portal-impl/src/com/liferay/portal/dao/jdbc/DataSourceFactoryImpl.java

```diff
@@ -620,6 +622,10 @@ protected void testDatabaseClass(String driverClassName) throws Exception {
 			}
 
 			try {
+				DigesterUtil digesterUtil = new DigesterUtil();
```

Comment:
@dantewang When will the NPE happen? I did not see it on startup.

Tina.

Reply:
@tinatian I got NPE when starting portal without this commit. `mysql.jar` is there in shielded-container-lib dir after running `ant all` so it needs to be removed for reproducing the NPE.